### PR TITLE
Account for root ownership of overlay files.

### PR DESCRIPTION
**Pull request by drewmoseley**
Since we chown and chmod files in the overlay, we need to ensure that we
can write to them in each of the bootstrap-rootfs-overlay generation
scripts.

We also need to build the docker image as root so that it can read
the files.

Eventually, I think we should make overlays as tarballs with ownership
and permissions inside the archive to avoid this dance.

Changelog: Title
Signed-off-by: Drew Moseley <drew.moseley@northern.tech>

# External Contributor Checklist

🚨 Please review the [guidelines for contributing](https://github.com/mendersoftware/mender/blob/master/CONTRIBUTING.md) to this repository.

- [ ] Make sure that all commits have a [`Changelog`](https://github.com/mendersoftware/mender/blob/master/CONTRIBUTING.md#changelog-tags) tag. If nothing should be added to the Changelog, add a `Changelog: None` tag. If there is a change, add `Changelog: Commit`, or `Changelog: Title`, depending on what should be included in the changelog.

- [ ] Make sure that all commits are signed with [`git --signoff`](https://git-scm.com/book/en/v2/Git-Tools-Signing-Your-Work). Also note that the signoff author must match the author of the commit.

### Description

Please describe your pull request.

Thank you!

### CODE REVIEW

**Comment by drewmoseley** [comment]
Hmm.  CI runner doesn't seem to have sudo?

**Comment by mirzak** [comment]
Can you elaborate a bit more on why this is needed? I am not really following and I do not understand how `sudo docker build` would change anything?

**Comment by lluiscampos** [comment]
> Hmm. CI runner doesn't seem to have sudo?

Add a line `- apk add sudo` to [build script](https://github.com/mendersoftware/mender-convert/blob/master/.gitlab-ci.yml#L37).

**Comment by drewmoseley** [comment]
Try to run any of the scripts a second time and they will fail due to the "chown" and "chmod" run the first time.

**Comment by drewmoseley** [comment]
Oh, also, with the chmod 600 of mender.conf, it cannot be read or installed unless docker-build is run as root.

**Comment by mirzak** [comment]
> also, with the chmod 600 of mender.conf, it cannot be read or installed unless docker-build is run as root.

This file is not needed to build the container, so just add it to `.dockerignore`, e.g:

```patch
$ git diff
diff --git a/.dockerignore b/.dockerignore
index 94e5f31..804e84e 100644
--- a/.dockerignore
+++ b/.dockerignore
@@ -1,3 +1,4 @@
 deploy
 input
 work
+rootfs_overlay_demo
```

File: .gitlab-ci.yml [hunk 1]
```diff
@@ -41,6 +41,10 @@ build:
     expire_in: 2w
     paths:
       - image.tar
+  before_script:
+    # Install dependencies
+    - apt update
+    - apt install -qyy sudo
```
Comment (lluiscampos) [inline .gitlab-ci.yml:47]: It is an Alpine Linux based image, we need `apk` instead of `apt-get`

Comment (drewmoseley) [reply .gitlab-ci.yml]: No longer needed after ignoring rootfs_overlay_*

**Comment by drewmoseley** [comment]
We still need the chown in the scripts.  Without it:
```
$ ./scripts/bootstrap-rootfs-overlay-hosted-server.sh  -o foo -t foo
Configuration file for using Hosted Mender written to: foo/etc/mender
$ ./scripts/bootstrap-rootfs-overlay-hosted-server.sh  -o foo -t foo
./scripts/bootstrap-rootfs-overlay-hosted-server.sh: line 57: foo/etc/mender/mender.conf: Permission denied
```

**Comment by drewmoseley** [comment]
@lluiscampos fixed

**Pull request merged**